static/app/views/onboarding/setupDocsLoader.tsx

```diff
@@ -0,0 +1,240 @@
+      }
+
+      setProjectKey(loadedKeys[0]);
+      setHasLoadingError(false);
+    } catch (error) {
+      setHasLoadingError(error);
+      throw error;
+    }
+  }, [api, organization.slug, project.slug]);
+
+  // Automatically update the products on the project key when the user changes the product selection
+  // Note that on initial visit, this will also update the project key with the default products (=all products)
+  // This DOES NOT take into account any initial products that may already be set on the project key - they will always be overwritten!
+  const updateSelectedProducts = useCallback(async () => {
+    const productsQuery =
+      (location.query.product as PRODUCT | PRODUCT[] | undefined) ?? [];
+    const products = typeof productsQuery === 'string' ? [productsQuery] : productsQuery;
+
+    const keyId = projectKey?.id;
+
+    if (!keyId) {
+      return;
+    }
+
+    const apiEndpoint = `/projects/${organization.slug}/${project.slug}/keys/${keyId}/`;
+
+    const newDynamicSdkLoaderOptions: Record<DynamicSDKLoaderOption, boolean> = {
+      [DynamicSDKLoaderOption.HAS_PERFORMANCE]: false,
+      [DynamicSDKLoaderOption.HAS_REPLAY]: false,
+      [DynamicSDKLoaderOption.HAS_DEBUG]: false,
+    };
+    products.forEach(product => {
+      // eslint-disable-next-line default-case
+      switch (product) {
+        case PRODUCT.PERFORMANCE_MONITORING:
+          newDynamicSdkLoaderOptions[DynamicSDKLoaderOption.HAS_PERFORMANCE] = true;
+          break;
+        case PRODUCT.SESSION_REPLAY:
+          newDynamicSdkLoaderOptions[DynamicSDKLoaderOption.HAS_REPLAY] = true;
+          break;
+      }
+    });
+
+    try {
+      await api.requestPromise(apiEndpoint, {
+        method: 'PUT',
+        data: {
+          dynamicSdkLoaderOptions: newDynamicSdkLoaderOptions,
+        },
+      });
+      setProjectKeyUpdateError(false);
+    } catch (error) {
+      const message = t('Unable to updated dynamic SDK loader configuration');
+      handleXhrErrorResponse(message)(error);
+      setProjectKeyUpdateError(true);
+    }
+  }, [api, location.query.product, organization.slug, project.slug, projectKey?.id]);
+
+  useEffect(() => {
+    fetchData();
+  }, [fetchData, organization.slug, project.slug]);
+
+  useEffect(() => {
+    updateSelectedProducts();
+  }, [updateSelectedProducts, location.query.product]);
+
+  return (
+    <Fragment>
+      <SetupIntroduction
+        stepHeaderText={t(
+          'Configure %s SDK',
+          platforms.find(p => p.id === currentPlatform)?.name ?? ''
+        )}
+        platform={currentPlatform}
+      />
+      <ProductSelection
+        defaultSelectedProducts={[PRODUCT.PERFORMANCE_MONITORING, PRODUCT.SESSION_REPLAY]}
+        lazyLoader
+        skipLazyLoader={close}
+      />
+
+      {projectKeyUpdateError && (
+        <LoadingError
+          message={t('Failed to update the project key with the selected products.')}
+          onRetry={() => updateSelectedProducts()}
```

Comment:
```suggestion
          onRetry={updateSelectedProducts}
```

nit: shall we name it `handleUpdateSelectedProducts`?

Reply:
makes sense! 👍